edualgo/backtracking_algorithms/n_queen.py

```diff
@@ -0,0 +1,187 @@
+# N-Queens Problem via Backtracking
+
+from __init__ import print_msg_box
+from time import time as ti
+n=0                                             #Size of the Board
+ans=[]                                          #Answer in the form of list of co-ordinates to place the Queens
+
+#Occupied Diagonals and Columns
+diagonal1={};diagonal2={}                       #For right and left Diagonal respectively
+Col={}                                          #For Column which are already alloted to some queen 
+
+def n_queens(board_size,hint=False):            #Main function argument =size of the board
```

Comment:
Place the main code inside a function such that any user can call the function and get the work done, for more details refer to the codebase and project documentations.

Reply:
the user can just call the function n_queens() with the size of the board as the argument and the work will be done